docs/adr/adr-11-blocksync-overhaul.md

```diff
@@ -0,0 +1,372 @@
+# ADR #011: Block Data Sync Overhaul: Part I - Storage
+
+## Changelog
+
+- 23.08.22: Initial unfinished draft
+- 14.09.22: The first finished version
+
+## Authors
+
+- @Wondertan
+
+> I start to like writing ADRs step-by-step. Also, there is a trick that helps: imagine like you are talking to a dev
+> who just joined the team to onboard him.
+
+## Glossary
+
+- LN - Light Node
+- FN - Full Node
+- BN - Bridge Node
+- [EDS(Extended Data Square)](https://github.com/celestiaorg/rsmt2d/blob/master/extendeddatasquare.go#L10) - plain Block
+data omitting headers and other block metadata.
+- [NMT](https://github.com/celestiaorg/nmt) - Namespaced Merkle Tree
+- [DataRoot](https://github.com/celestiaorg/celestia-node/blob/main/service/share/share.go#L35) - alias for
+[DAHeader](https://github.com/celestiaorg/celestia-core/blob/v0.34.x-celestia/pkg/da/data_availability_header.go#L29)
+
+## Context
+
+### Status Quo
+
+Current block data synchronization is done over Bitswap, traversing NMT trees of rows and columns of data square quadrants.
+We know from empirical evidence that it takes more than 200 seconds(~65000 network requests) to download a 4MB block of
+256kb shares, which is unacceptable and must be much less than the block time(15/30sec).
+
+The DASing, on the other hand, shows acceptable metrics for the block sizes we are aiming for initially. In the case of
+the same block, a DAS operation takes 50ms * 8(technically 9) blocking requests, which is ~400ms in an ideal scenario
+(excluding disk IO).
```

Comment:
This may sounds as if we won't be touching/improving DASing at all. Though my understandig is that why it is acceptable initially, we still will want to optimize DASing as well (I think that is sth @derrandz will look into among other things AFIU).
Nothing to change here but just double-checking that we are aligned.